### PR TITLE
Deividas develop

### PR DESCRIPTION
**Pull request by De1xs**
AddCategory : adding functionality to add your own transaction categories

Transactions: Added date field to track date and time

Statistics: Added statistics service to track statistics (any chosen range of start and end dates)
Added additional tab in main UI to show these statistics
**NEEDS UI FIX**

### CODE REVIEW

File: CategoryService.cs [hunk 1]
```diff
@@ -23,9 +23,13 @@ public string[] GetCategories()
             TextFileReader reader = new TextFileReader();
             string[] data = reader.FetchStringArrayByLocation(System.AppDomain.CurrentDomain.BaseDirectory + @"..\..\Data\Categories.txt");
             if (data == null) return null;
-            string[] categories = new string[data.Length];
+            string[] categories = new string[data.Length-1];
```
Comment (AivarasPet) [inline CategoryService.cs:26]: why s there data.lenght - 1?

File: CategoryService.cs [hunk 2]
```diff
@@ -23,9 +23,13 @@ public string[] GetCategories()
...
             for (int x = 0; x < data.Length; x++)
             {
+                if (data[x] == "")
+                {
+                    continue;
+                }
```
Comment (AivarasPet) [inline CategoryService.cs]: fit it in one line, pls, wastes space

File: Statistics/StatisticsService.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                        income += t.Amount;
+                    }
+                    else
+                    {
+                        expenses += t.Amount;
+                    }
+
+                    
+                    if (result.TryGetValue(t.Category, out int count))
+                    {
```
Comment (AivarasPet) [inline Statistics/StatisticsService.cs]: Too many large if else statements

**Review by AivarasPet — CHANGES_REQUESTED** [review]
Change stuff D:

File: CategoryService.cs (hunk 1 above)
Comment (De1xs) [reply CategoryService.cs:26]: kad nebutu tuscio optiono

**Comment by De1xs** [comment]
Nu kaip dabar vaikai

File: CategoryService.cs (hunk 2 above)
Comment (De1xs) [reply CategoryService.cs]: done

File: Statistics/StatisticsService.cs (hunk 3 above)
Comment (De1xs) [reply Statistics/StatisticsService.cs]: done

**Pull request merged**